Engines/Wine/Verbs/D9VK/script.js

```diff
@@ -55,35 +64,24 @@ class D9VK {
                     // choice: "Exit D9VK Installer, but continue with the script"
                     return this;
                 default:
-                    // do nothing
+                // do nothing
             }
         }
-        else
-        {
+        else {
             wizard.message(tr("Please ensure you have the latest drivers (418.30 minimum for NVIDIA and mesa 19 for AMD) or else D9VK might not work correctly."));
         }
 
-        if (typeof this.d9vkVersion !== 'string')
-        {
-            const versions = getGithubReleases("Joshua-Ashton", "d9vk", wizard);
-            this.d9vkVersion = versions[0];
-        }
+        const selectedVersion = this.selectGithubVersion(wizard);
 
-        var setupFile = new Resource()
-            .wizard(wizard)
-            .url(
-                `https://github.com/Joshua-Ashton/d9vk/releases/download/${this.d9vkVersion}/d9vk-${this.d9vkVersion}.tar.gz`
-            )
-            .name(`d9vk-${this.d9vkVersion}.tar.gz`)
-            .get();
+        var setupFile = this.downloadGithubRelease(selectedVersion, wizard);
 
         new Extractor()
             .wizard(wizard)
             .archive(setupFile)
             .to(`${prefixDirectory}/TMP/`)
             .extract();
 
-        const d9vkTmpDir = `${prefixDirectory}/TMP/d9vk-${this.d9vkVersion}`;
+        const d9vkTmpDir = `${prefixDirectory}/TMP/d9vk-${selectedVersion.name}`;
```

Comment:
We have this pattern in multiple places and I don't like it. The version name should not be needed in the folder names in my opinion, especially because we have no "safe" method to fetch it. Who guarantees that `name` or `tag_name` contains the correct version name?

Reply:
Yes, also it does not seem to be required at all.

Reply:
These folder names seem to be dependent on the content of the downloaded archives. I don't think we can replace them with a real temporary directory.